### PR TITLE
Postgis provider support

### PR DESCRIPTION
**Pull request by jorgejesus**
PostGIS support using generic SQL in-sync queries using psycopg2 library

Provider supports schema and implements server side cursors for item pagination

### CODE REVIEW

File: openapi.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+[2018-09-07T20:37:57Z] {/home/jesus/git/pygeoapi/pygeoapi/provider/elasticsearch_.py:69} ERROR - Cannot connect to Elasticsearch
```
Comment (tomkralidis) [inline openapi.yml]: Is this file needed?

File: pygeoapi-openapi.yml [hunk 2]
```diff
@@ -0,0 +1,273 @@
+components:
```
Comment (tomkralidis) [inline pygeoapi-openapi.yml]: Is this file needed also? We have a utility to generate this already.

File: pygeoapi/provider/__init__.py [hunk 3]
```diff
@@ -37,7 +37,8 @@
     'Elasticsearch': 'pygeoapi.provider.elasticsearch_.ElasticsearchProvider',
     'GeoJSON': 'pygeoapi.provider.geojson.GeoJSONProvider',
     'GeoPackage': 'pygeoapi.provider.geopackage.GeoPackageProvider',
-    'SQLite': 'pygeoapi.provider.sqlite.SQLiteProvider'
+    'SQLite': 'pygeoapi.provider.sqlite.SQLiteProvider',
+    'PostgreSQL': 'pygeoapi.provider.postgresql.PostgreSQLProvider'
```
Comment (tomkralidis) [inline pygeoapi/provider/__init__.py]: Can we try to keep to alphabetical order?

File: pygeoapi/provider/postgresql.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        :returns: GeoJSON FeaturesCollection
+        """
+        LOGGER.debug('Querying PostGIS')
+
+        if resulttype == 'hits':
+
+            with DatabaseConnection(self.conn_dic,
+                                    self.table, context="hits") as db:
+                cursor = db.conn.cursor(cursor_factory=RealDictCursor)
+                sql_query = SQL("select count(*) as hits from {}").\
```
Comment (tomkralidis) [inline pygeoapi/provider/postgresql.py]: Can we implement binding here to make safer against SQL injections?

Comment (jorgejesus) [reply pygeoapi/provider/postgresql.py]: Identifier(self.table) on line 182 should take care of the SQL injection as indicated in http://initd.org/psycopg/docs/sql.html  as explained on class psycopg2.sql.SQL(string) and on  class psycopg2.sql.Identifier(*strings)
